apps/remote_control/lib/lexical/remote_control/search/store.ex

```diff
@@ -66,34 +70,34 @@ defmodule Lexical.RemoteControl.Search.Store do
     GenServer.call(__MODULE__, {:unique_fields, fields})
   end
 
-  @spec start_link(Project.t(), create_index, refresh_index) :: GenServer.on_start()
-  def start_link(%Project{} = project, create_index, refresh_index) do
-    start_link([project, create_index, refresh_index])
+  @spec start_link(Project.t(), create_index, update_index) :: GenServer.on_start()
+  def start_link(%Project{} = project, create_index, update_index) do
+    start_link([project, create_index, update_index])
   end
 
-  def start_link([%Project{} = project, create_index, refresh_index]) do
-    GenServer.start_link(__MODULE__, [project, create_index, refresh_index], name: __MODULE__)
+  def start_link([%Project{} = project, create_index, update_index]) do
+    GenServer.start_link(__MODULE__, [project, create_index, update_index], name: __MODULE__)
   end
 
-  def start_link([create_index, refresh_index]) do
-    start_link(Lexical.RemoteControl.get_project(), create_index, refresh_index)
+  def start_link([create_index, update_index]) do
+    start_link(Lexical.RemoteControl.get_project(), create_index, update_index)
   end
 
   def init([%Project{} = project, create_index, update_index]) do
-    state = State.new(project, create_index, update_index)
+    state =
+      project
+      |> State.new(create_index, update_index)
+      |> State.async_load()
 
-    {:ok, state, {:continue, :load}}
+    {:ok, state}
   end
 
-  def handle_continue(:load, %State{} = state) do
-    case State.load(state) do
-      {:ok, state} ->
-        {:noreply, state}
+  def handle_info({ref, result}, %State{async_load_ref: ref} = state) do
```

Comment:
put a commnt here that this is coming from `Task`. You also need to handle the possible `DOWN` message mentioned here: https://hexdocs.pm/elixir/1.13/Task.html#await/2-compatibility-with-otp-behaviours

Reply:
Okay so handling `DOWN` is actually a little tricky and may not be necessary.

Since I'm using `Task.async/1` to start the tasks, the caller is linked to them. That means that if the task *crashes*, the store is going to crash. (Note that this retains the same behavior as before the PR, since the `create_index` and `update_index` functions were just being run directly in the store process.)

So the only `DOWN` messages we'll be able to receive are those where the task completed, which means we'll receive the result and handle it.

If we *don't* want a crashed task to crash the store, we'll have to start them in a `Task.Supervisor`, which lets us use `Task.Supervisor.async_nolink`. I don't think that's necessarily a bad idea, but it may be something that can be addressed later and not in this PR.

Thoughts?